src/Storefront/Pagelet/Wishlist/GuestWishlistPageletLoader.php

```diff
@@ -38,11 +39,12 @@ public function __construct(
     public function load(Request $request, SalesChannelContext $context): GuestWishlistPagelet
     {
         $page = new GuestWishlistPagelet();
+        $criteria = new Criteria();
 
-        $criteria = $this->createCriteria($request, $context);
-        $this->eventDispatcher->dispatch(new GuestWishListPageletProductCriteriaEvent($criteria, $context, $request));
-
-        if (empty($criteria->getIds())) {
+        try {
+            $criteria = $this->createCriteria($request, $context);
+            $response = $this->productListRoute->load($criteria, $context);
+        } catch (DataAbstractionLayerException $e) {
```

Comment:
why did you changed it like this? the logic is different now. 

we should probably fix the throwing of this exception instead

Reply:
@mitelg I think it's the same. Because now `setIds()` on criteria throws an exception.

Reply:
In this case the code specifically checked for the ids being empty, so i assumed it was expected behaviour. Therefore it seemed to be most correct to run the code behind that check when instantiating the criteria object fails.

I do however not know much about the code.

Reply:
first the criteria is created and then the event is dispatched. that is now changed to dispatching the event after the loading of the product list. which is wrong

Reply:
ah yeah, thanks. I will take a look.

Reply:
it's a bit tricky because we no longer have a valid `Criteria` object to dispatch `GuestWishListPageletProductCriteriaEvent` with

Reply:
Should i do something here or will you handle it?

Reply:
@Benedikt-Brunner I made the changes and its internal review now. Will ping you when it's merged :)